GoogleUtilities/Example/Tests/Logger/GULLoggerTest.m

```diff
@@ -12,184 +12,88 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifdef DEBUG
-// The tests depend upon library methods only built with #ifdef DEBUG
-
 #import <OCMock/OCMock.h>
 #import <XCTest/XCTest.h>
 
 #import <GoogleUtilities/GULLogger.h>
 
-#import <asl.h>
-
-extern const char *kGULLoggerASLClientFacilityName;
-
-extern void GULResetLogger(void);
-
-extern aslclient getGULLoggerClient(void);
-
-extern dispatch_queue_t getGULClientQueue(void);
+static GULLoggerLevel kLogLevel = GULLoggerLevelError;
+static NSString *const kService = @"Test Service";
+static NSString *const kCode = @"I-COR000001";
+static NSString *const kLogMessage = @"Log Message";
+static NSString *const kVersionString = @"2";
+static char *const kVersionChar = "2";
 
-extern BOOL getGULLoggerDebugMode(void);
+// Redefine class property as readwrite for testing.
+@interface GULLogger (ForTesting)
+@property(nonatomic, class, readwrite) id<GULLoggerSystem> logger;
+@end
 
-static NSString *const kMessageCode = @"I-COR000001";
+#pragma mark -
 
 @interface GULLoggerTest : XCTestCase
-
-@property(nonatomic) NSString *randomLogString;
-
-@property(nonatomic, strong) NSUserDefaults *defaults;
-
+@property(nonatomic) id loggerSystemMock;
 @end
 
 @implementation GULLoggerTest
 
 - (void)setUp {
-  [super setUp];
-  GULResetLogger();
-
-  // Stub NSUserDefaults for cleaner testing.
-  _defaults = [[NSUserDefaults alloc] initWithSuiteName:@"com.google.logger_test"];
+  self.loggerSystemMock = OCMProtocolMock(@protocol(GULLoggerSystem));
+  GULLogger.logger = self.loggerSystemMock;
 }
 
 - (void)tearDown {
-  [super tearDown];
-
-  _defaults = nil;
+  GULLogger.logger = nil;
```

Comment:
It's also helpful to call `[self.loggerSystemMock stopMocking]` just to avoid any conflicts. I know it's probably redundant here, but I've seen `OCMock` do odd things at times.

Reply:
Done. I also called `stopMocking` on any other mocks from the tests I touched.